webapp/components/login/login_controller.jsx

```diff
@@ -60,9 +60,21 @@ export default class LoginController extends React.Component {
             GlobalActions.redirectUserToDefaultTeam();
         }
 
+        if (this.props.location.query.extra === Constants.SIGNIN_VERIFIED && this.props.location.query.email) {
+            this.refs.password.focus();
+        }
+
         AsyncClient.checkVersion();
     }
 
+    componentWillMount() {
```

Comment:
`componentWillMount` is equivalent to the constructor. Let's put this code there.